app/_plugins/filters/slugify.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module SlugifyFilter
+    def slugify(input)
+      if input.is_a? String
+        input.downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '')
+      else
+        raise "Invalid input type: #{input.class} to 'slugify'. Input must be a string."
+      end
```

Comment:
@mheap jekyll has a built-in [slugify filter](https://jekyllrb.com/docs/liquid/filters/) :)